compile error on typescript@next

### Version
3.0.0-alpha.7

### Reproduction link
[https://github.com/microsoft/TypeScript/pull/37225#discussion_r388423749](https://github.com/microsoft/TypeScript/pull/37225#discussion_r388423749)







### Steps to reproduce
Upgrade to typescript@next, then run build or open packages/reactivity/src/collectionHandlers.ts:179 with your editor set to use the local version of typescript.

### What is expected?
No compile error

### What is actually happening?
Compile error: semantic error TS2784: 'get' and 'set' accessors cannot declare 'this' parameters.

---
Typescript did not intend to allow 'this' parameters for accessors and wasn't checking them, but incorrectly allowed them until Microsoft/Typescript#36889. Discovered by Typescript's user test run: https://github.com/microsoft/TypeScript/pull/37225#discussion_r388423749

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -176,8 +176,8 @@ const mutableInstrumentations: Record<string, Function> = {
   get(this: MapTypes, key: unknown) {
     return get(this, key, toReactive)
   },
-  get size(this: IterableCollections) {
-    return size(this)
+  get size() {
+    return size(this as unknown as IterableCollections)
   },
   has,
   add,

